### ISSUE TITLE
protection against self-assignment

### ISSUE THREAD

**Issue by gitavk** [PyCQA/pylint#2930]
Based on the [SO question](https://stackoverflow.com/questions/56112506/pylint-protection-again-self-assignment) it can be good point to add the default check for the self-assignment.
For example
```
class Aclass:
    def __init__(self, attr=None, attr2=None):
        self.attr = attr
        self.attr2 = attr2

def new_a(attr, attr2):
    a_inst = Aclass()
    a_inst.attr2 = attr2
    # should be: a_inst.attr = attr, but have a typo
    attr = attr
    return a_inst
```

**Comment by PCManticore** [comment]
Thanks, this would be good to have!

**Comment by chaoflow** [comment]
@PCManticore @gitavk The new check catches the following code as erroneous:

```python
FOO = 1

class Bar:
    FOO = FOO
```

Is this intentional?

**Comment by PCManticore** [comment]
Looks like an untreated edge case to me, thanks for the ping.

**Comment by chaoflow** [comment]
@PCManticore Thank you for the quick fix!
